Don't define assorted things for userspace.  In particular do not define
inline to __attribute__((always_inline)).

--- include/linux/compiler.h	2003-10-15 11:13:09.000000000 -0400
+++ include/linux/compiler.h.t	2003-11-01 18:04:19.000000000 -0500
@@ -9,6 +9,15 @@
 # define __kernel
 #endif
 
+#if !defined(__KERNEL__)
+/* Debian: Most of these are inappropriate for userspace.  */
+/* We don't define likely, unlikely, or barrier; they're namespace-intrusive
+   and should not be needed outside of __KERNEL__.  */
+# define __attribute_pure__
+# define __attribute_used__
+# define __deprecated
+#else
+
 #if __GNUC__ > 3
 # include <linux/compiler-gcc+.h>	/* catch-all for GCC 4, 5, etc. */
 #elif __GNUC__ == 3
@@ -86,4 +95,6 @@
     (typeof(ptr)) (__ptr + (off)); })
 #endif
 
+#endif /* __KERNEL__ */
+
 #endif /* __LINUX_COMPILER_H */
